trin-core/src/portalnet/overlay_service.rs

```diff
@@ -933,20 +934,26 @@ where
                 ))
             })?;
 
-            requested_keys
-                .set(
-                    i,
-                    self.storage.read().should_store(&key).map_err(|err| {
```

Comment:
In this case, `should_store()` seems preferable to encapsulate the [checking radius logic and checking if it's already been stored logic] at the storage layer.

Reply:
The disadvantage of `should_store` is that the reason you should or should not store some data is opaque. You receive a boolean, a yes or no, but you do not know why. It seems useful to be able to differentiate between the radius check and the contains check. You could return an enum from `should_store`, but then you would need to do some matching and end up with something similar to what I have below.

This combination of logic would be less clunky without the `map_err`. A `DataStoreError` (or similar) should be added so that we can use the `?` syntax directly after `get`.

Reply:
Just a standard word of caution, since it's something I'm guilty of all the time, but just b/c "It seems useful to be able to differentiate between the radius check and the contains check.", doesn't mean we need to introduce this complexity until it's actually required. Until we have a valid reason for knowing whether data wasn't stored b/c it already exists, or b/c it falls outside our radius, why not just try and push it in and ignore the result. But, maybe there is a valid reason use case now/down the road i'm missing

Reply:
@njgheorghita well my argument is that it's not additional complexity, and that it could be considered to be removal of complexity. To me, it makes the storage behavior easier to reason about. The opaque and more complicated `should_store` function is broken down into simple components. Either way, we can still call `put` to attempt to store the data and ignore the result.